students/Jean-Baptiste/lessons/lesson01/assignment/assign_01_inventory/test_unit.py

```diff
@@ -0,0 +1,283 @@
+        self.assertEqual(f.check_for_big_screen(), True)
+
+
+class Test_InventoryClass(TestCase):
+    def test_init(self):
+        item_dict = {}
+        item_dict["item_code"] = 1
+        item_dict["item_description"] = "a"
+        item_dict["item_rental_price"] = 1.0
+        item_dict["item_price"] = 1.0
+        i = Inventory(**item_dict)
+        self.assertEqual(i.product_code, 1)
+        self.assertEqual(i.description, "a")
+        self.assertEqual(i.rental_price, 1.0)
+        self.assertEqual(i.market_price, 1.0)
+
+    def test_return_as_dictionary(self):
+        item_dict = {}
+        item_dict["item_code"] = 1
+        item_dict["item_description"] = "a"
+        item_dict["item_rental_price"] = 1.0
+        item_dict["item_price"] = 1.0
+        i = Inventory(**item_dict)
+        new_dict = i.return_as_dictionary()
+        self.assertEqual(i.product_code, new_dict["product_code"])
+        self.assertEqual(i.description, new_dict["description"])
+        self.assertEqual(i.rental_price, new_dict["rental_price"])
+
+    def test_check_market_price(self):
+        item_dict = {}
+        item_dict["item_code"] = 1
+        item_dict["item_description"] = "a"
+        item_dict["item_rental_price"] = 1.0
+        item_dict["item_price"] = 1.0
+        i = Inventory(**item_dict)
+        self.assertEqual(i.check_market_price(), False)
+        i.market_price = get_latest_price()
+        self.assertEqual(i.check_market_price(), True)
+
+
+class Test_Main(TestCase):
+    def test_generate_menu(self):
+        self.assertEqual(generate_menu("Z"), False)
+        self.assertEqual(generate_menu("1").__name__, "add_new_item")
+        self.assertEqual(generate_menu("2").__name__, "item_info")
+        self.assertEqual(generate_menu("q").__name__, "exit_program")
+
+    def test_get_price(self):
+        self.assertEqual(get_price(), get_latest_price())
+
+    def test_return_item_info(self):
+        item_dict = {}
+        item_dict["item_code"] = 1
+        item_dict["item_description"] = "a"
+        item_dict["item_rental_price"] = 1.0
+        item_dict["item_price"] = 1.0
+        i1 = Inventory(**item_dict)
+        item_dict1 = {}
+        item_dict1["item_code"] = 2
+        item_dict1["item_description"] = "b"
+        item_dict1["item_rental_price"] = 2.0
+        item_dict1["item_price"] = 2.0
+        i2 = Inventory(**item_dict1)
+        f_inventory = {}
+        f_inventory["1"] = i1.return_as_dictionary()
+        f_inventory["2"] = i2.return_as_dictionary()
+        self.assertEqual(
+            return_item_info(f_inventory, "1"),
+            "product_code:1\ndescription:a\nmarket_price:1.0\nrental_price:1.0\n"
+        )
+        self.assertEqual(
+            return_item_info(f_inventory, "2"),
+            "product_code:2\ndescription:b\nmarket_price:2.0\nrental_price:2.0\n"
+        )
+        self.assertEqual(
+            return_item_info(f_inventory, "z"), "Item not found in inventory")
```

Comment:
You should test just one thing at a time in unit tests. i1, i2, and f_inventory should be in separate tests. Possibly using parameterize for doing 3 in 1. https://pypi.org/project/parameterized/